skt/reporter.py

```diff
@@ -222,38 +250,28 @@ def getfulllog(self):
 
     def gettraces(self):
         """
-        Get a list of oops and call stack outputs extracted from the kernel
-        console log.
+        Get a list of non-overlapping oops and call stack outputs extracted
+        from the kernel console log.
 
         Returns:
             A list of oops and call stack output strings.
         """
         result = []
-
-        insplat = False
-        inct = False
         tmpdata = []
-        # FIXME Check if line == True otherwise it adds an empty line at the
-        # end of the extracted trace.
+
         for line in self.data:
-            if self.oopspattern.search(line):
-                insplat = True
-            elif re.search("Call Trace:", line):
-                inct = True
-
-            if insplat and ((inct and not self.ctvpattern.search(line)) or
-                            self.eendpattern.search(line)):
-                tmpdata.append(line)
-                result.append("\n".join(tmpdata))
-                tmpdata = []
-                insplat = False
-                inct = False
-
-            if insplat:
-                tmpdata.append(line)
-
-        if tmpdata:
-            result.append("\n".join(tmpdata))
+            if self.invalid_pattern.search(line):
+                continue
+            if self.start_pattern.search(line):
+                tmpdata = [line]
+            elif tmpdata:
+                if self.end_pattern.search(line):
+                    tmpdata.append(line)
+                    result.append('\n'.join(tmpdata))
+                    tmpdata = []
+                    continue
+                if self.continue_pattern.search(line):
+                    tmpdata.append(line)
```

Comment:
I am trying to understand this logic.  If we see a start pattern, start recording.  Makes sense.  If we see an end pattern, append final line, record to results, and reset tmpdata.  Makes sense.

I get lost on the 'if continue pattern' part.  It seems like we would randomly drop lines between start and end pattern if they do not match 'continue pattern'?  Basically what happens when 'if continue_pattern' fails?